### PR TITLE
Create ansible playbook to deploy the pygrid

### PR DESCRIPTION
**Pull request by javanlacerda**
# Pull Request Template

## Description

Create Ansible playbook to deploy PyGrid components in remote hosts.

Fixes # (#407 )

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- [X] Running in Ubuntu 18.04 environment, using a OpenStack instance (Ubuntu 16.04) as remote host

## Checklist:

- [X] I did follow the [contribution guidelines](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md)
- [ ] New Unit tests added
- [ ] Unit tests pass locally with my changes

### CODE REVIEW

File: ansible/pygrid_deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+- hosts: hosts
```
Comment (kaiokmo) [inline ansible/pygrid_deploy.yaml]: you might want to start (first line) the yaml file with `---`
also, you should also include a `name` on your playbook

File: ansible/pygrid_deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      register: output
+      changed_when: output.stdout
+    - setup: # aka gather_facts
+  
+  tasks:
+    - name: Update apt
+      apt:
+        upgrade: yes
+        update_cache: yes
+      become: true
```
Comment (kaiokmo) [inline ansible/pygrid_deploy.yaml]: I'd rather leave privilege escalations to be configured on `ansible.cfg` than having `become: true` on every task, and then you could specify on `README.md` what should be configured

File: ansible/pygrid_deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      apt:
+        name: python-pip
+      become: true
+
+    - name: Install python docker
+      pip:
+        name: docker-py
+      become: true
+    
+    - docker_container:
```
Comment (kaiokmo) [inline ansible/pygrid_deploy.yaml]: you should also give `name` to all the tasks to improve readability and debugging

File: ansible/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# Ansible playbook to deploy PyGrid components (for Debian based systems)
+
+## Install ansible
+
+Before you try run any ansible script, you need install ansible. You can find more
```
Comment (kaiokmo) [inline ansible/README.md]: `Before running ansible, it needs to be installed. One can find more` ...

File: ansible/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+information about installation to your linux distribution [here](http://docs.ansible.com/ansible/intro_installation.html).
+
+If you use Debian based distribution, you can install using the commands below:
```
Comment (kaiokmo) [inline ansible/README.md]: The `README` already mentions Debian-based distributions. I'd rather remove this mention here and add instructions of how to also install `ansible` as a pip package under a `virtualenv`.

File: ansible/README.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+```
+$ apt-get update
+$ apt-get install ansible
+```
+
+### Editing ansible.cfg to not ask ssh key host checking
+
+SSH always ask if you sure in establish a ssh connection in your first time that you try to connect in a machine. For jump this, ucomment the line in **/etc/ansible/ansible.cfg**:
```
Comment (kaiokmo) [inline ansible/README.md]: there are some typos here

File: ansible/README.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+```
+host_key_checking = False
```
Comment (kaiokmo) [inline ansible/README.md]: you could mention that this should be added under the `defaults` section

File: ansible/README.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+## Add keys in machine that will be managed
+
+To ansibe install and configure machine, it is necessary that your key is added in
```
Comment (kaiokmo) [inline ansible/README.md]: typo

File: ansible/README.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+**.ssh/authorized_keys** in machine that will be managed. To do this, run the command
+below:
+
+```
+$ ssh-copy-id -i ~/<route_ssh_public_key> <user>@<ip_or_hostname>
+```
+
+## Add hosts to the inventory
+
+Before all, add the hosts that will recieve the containers to inventory file like:
```
Comment (kaiokmo) [inline ansible/README.md]: `Before anything...`, and also a typo on `recieve`

**Comment by hericlesme** [comment]
@kaiokmo Thanks for the review! <3

**Comment by javanlacerda** [comment]
Thanks @kaiokmo for reviewing! All checks solved!

**Comment by kaiokmo** [comment]
+1

**Comment by hericlesme** [comment]
Awesome!

**Pull request merged**